api/api_test.go

```diff
@@ -538,3 +539,36 @@ func TestCloneHttpClient(t *testing.T) {
 	})
 
 }
+
+func TestClient_HeaderRaceCondition(t *testing.T) {
```

Comment:
I feel a bit odd about this test: `-race` can detect the condition deterministically I suspect without needing to  thousands of iterations. Also, it's probably sufficient to check that `conf.Headers` aren't mutated.

Reply:
Yeah, I struggled to get `conf.Headers` to be mutated. I _think_ that's because the underlying map was being reallocated, but wasn't able to easily validate because of the effort that go expends to pretend that `map` isn't a pointer.

Reply:
Will look into `-race` testing.